library/Pdfexport/HeadlessChrome.php

```diff
@@ -247,106 +264,169 @@ public function fromHtml($html, $asFile = false)
+                    $killer = Loop::addTimer(10, function (TimerInterface $timer) use ($chrome, $deferred) {
+                        $chrome->terminate(6); // SIGABRT
+
+                        $interval = $timer->getInterval();
+                        // Reject/resolve the promise on its own event tick not to block the current one.
+                        Loop::futureTick(function () use ($deferred, $interval) {
+                            $deferred->reject(
+                                new HeadlessChromeException(
+                                    sprintf(
+                                        'Browser timed out after %d seconds without the expected output',
+                                        $interval
+                                    )
+                                )
                             );
-                        } catch (Exception $e) {
-                            Logger::error('Failed to print PDF. An error occurred: %s', $e);
+                        });
+                    });
+
+                    $chrome->start();
+
+                    $chrome->stderr->on('data', function ($chunk) use ($chrome, $deferred, $killer) {
+                        Logger::debug('Caught browser output: %s', $chunk);
+
+                        if (preg_match(self::DEBUG_ADDR_PATTERN, trim($chunk), $matches)) {
+                            Loop::cancelTimer($killer);
+
+                            $printParams = ! $this->document->isEmpty() ? $this->document->getPrintParameters() : [];
+
+                            try {
+                                $pdf = $this->printToPDF($matches[1], $matches[2], $printParams);
+                                // Reject/resolve the promise on its own event tick not to block the current one.
+                                Loop::futureTick(function () use ($deferred, $pdf) {
+                                    if (! empty($pdf)) {
+                                        $deferred->resolve($pdf);
+                                    } else {
+                                        $deferred->reject(
+                                            new Exception(
+                                                'Received empty response or none at all from browser.'
+                                                . ' Please check the logs for further details.'
+                                            )
+                                        );
+                                    }
+                                });
+                            } catch (Exception $e) {
+                                // Reject/resolve the promise on its own event tick not to block the current one.
+                                Loop::futureTick(function () use ($e, $deferred) {
+                                    $deferred->reject(
+                                        new HeadlessChromeException(
+                                            'Failed to print PDF. An error occurred: ',
```

Comment:
Log this, as discussed. Afterwards, `HeadlessChromeException` should be redundant and removed.